controllers/nginx/rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -221,10 +221,11 @@ http {
         {{ $path := buildLocation $location }}
         {{ $authPath := buildAuthLocation $location }}
 
-        {{ if not (empty $location.CertificateAuth.CertFileName) }}
+        {{ if not (empty $location.CertificateAuth.CAFileName) }}
         # PEM sha: {{ $location.CertificateAuth.PemSHA }}
         ssl_client_certificate              {{ $location.CertificateAuth.CAFileName }};
         ssl_verify_client on;
+        ssl_verify_depth 10;
```

Comment:
2 is the most common depth for normal public-ca-minted certs, though 3 isn't uncommon (e.g. cross-signed certs, CAs that have second layers of intermediates).
Out of the alexa top 100 sites, 80% are depth=2, 9% depth=1, and 11% didn't feel like responding on 443.

However, for client cert auth, public CAs wouldn't generally be used. In that case, 2 and 1 are both quite common. 1 is common for organizations that have a root CA and directly issue client certs from that, while organizations that try to practice something akin to what public CAs do (e.g. keeping the root CA in offline storage and signing from intermediates on HSMs) will have a depth of 2.

It's worth noting that firefox, iirc, has a max verify_depth of 8 before it gives up.